test/letters/containers/VeteranBenefitSummaryLetter.unit.spec.jsx

```diff
@@ -51,52 +46,93 @@ describe('<VeteranBenefitSummaryLetter>', () => {
 
   it('should show service info options', () => {
-    const tree = SkinDeep.shallowRender(<VeteranBenefitSummaryLetter store={store} {...defaultProps}/>);
-    expect(tree.dive(['div', 'div', '#militaryService'])).not.to.be.empty;
+    const tree = SkinDeep.shallowRender(<VeteranBenefitSummaryLetter {...defaultProps}/>);
+    expect(tree.subTree('#militaryService')).not.to.be.empty;
+  });
+
+  it('renders error and hides benefit table if options not available', () => {
+    const props = _.set('optionsAvailable', false, defaultProps);
+    const tree = SkinDeep.shallowRender(<VeteranBenefitSummaryLetter {...props}/>);
+    const headerText = tree.dive(['.feature', 'h4']).text();
+    expect(headerText).to.equal('Your VA Benefit Summary letter is currently unavailable');
+    expect(tree.subTree('#benefitInfoTable')).to.be.false;
   });
 
-  it('should update request option when checked', () => {
-    const oldWindow = global.window;
+  it('maps each service entry to its own table row', () => {
+    const navyService = [{
+      branch: 'NAVY',
+      characterOfService: 'UNCHARACTERIZED',
+      enteredDate: '1974-01-01T05:00:00.000+00:00',
+      releasedDate: '1976-10-01T04:00:00.000+00:00'
+    }];
+
+    const doubleService = defaultProps.benefitSummaryOptions.serviceInfo.concat(navyService);
+    const props = _.set('benefitSummaryOptions.serviceInfo', doubleService, defaultProps);
+    const tree = SkinDeep.shallowRender(<VeteranBenefitSummaryLetter {...props}/>);
+    const serviceRows = tree.dive(['#militaryServiceTable', 'tbody']).everySubTree('tr');
+
+    expect(serviceRows.length).to.equal(doubleService.length);
+  });
+
+  it('handles check and uncheck events', () => {
+    const oldWindow = { ...global.window };
     global.window = {
       dataLayer: [],
     };
 
-    const updateOption = sinon.spy();
-    const props = _.set('updateBenefitSummaryRequestOption', updateOption, defaultProps);
+    const updateOptionSpy = sinon.spy();
+    const props = _.set('updateBenefitSummaryRequestOption', updateOptionSpy, defaultProps);
     const component = ReactTestUtils.renderIntoDocument(
-      <VeteranBenefitSummaryLetter store={store} {...props}/>
+      <VeteranBenefitSummaryLetter {...props}/>
     );
 
+    // sanity check
+    expect(global.window.dataLayer.length).to.equal(0);
+
     const formDOM = findDOMNode(component);
     const inputs = formDOM.querySelectorAll('input');
+
+    // one input for service info, two more for benefitInfo.*
     expect(inputs.length).to.equal(3);
 
-    ReactTestUtils.Simulate.change(inputs[0], {
-      target: {
-        checked: false
-      }
-    });
-    ReactTestUtils.Simulate.change(inputs[1], {
+    const DOMid = 'militaryService';
+    const checkedValue = false;
+
+    const mockDOMEvent = {
       target: {
-        checked: false
-      }
-    });
+        id: DOMid,
+        checked: checkedValue,
+      },
+    };
+
+    ReactTestUtils.Simulate.change(inputs[0], mockDOMEvent);
```

Comment:
If you use `getFormDOM` from the `test/util/schemaform-utils`, we have a `setCheckbox` method to shorten this.

Reply:
Ahh, good to know! I'm just gonna keep this version only because I'm also testing the contents of `global.window.dataLayer`, which has an `id` property for each item in the array - using `formDOM` doesn't appear to support specifying an id property for the event.